decoder.go

```diff
@@ -472,7 +462,9 @@ func (d *decoder) textUnmarshaller(key string, value reflect.Value, str string)
 		return nil
 	}
 
-	if value.IsNil() {
+	kind := value.Kind()
+
+	if kind >= reflect.Array && kind <= reflect.Slice && value.IsNil() {
```

Comment:
This was the most concise way I could make this expression: https://golang.org/src/reflect/type.go#L244 . Another option would be to have 6 comparisons.. This is needed only because of the check for nil will panic if it is called on other than these types.